ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/Client.java

```diff
@@ -311,6 +312,146 @@ CompletableFuture<Void> createConnector(
    */
   CompletableFuture<ConnectorDescription> describeConnector(String connectorName);
 
+  /**
+   * Asserts the existence of a schema with the given subject name. If the assertion fails,
+   * then the {@code CompletableFuture} will complete exceptionally
+   *
+   * <p>If a non-200 response is received from the server, the {@code CompletableFuture} will be
+   * failed.
+   *
+   * @param subject subject name
+   * @param exists true to check that a schema exists, false to check that a schema does not exist
+   * @return a future that completes once the server response is received
+   */
+  CompletableFuture<Void> assertSchema(String subject, boolean exists);
```

Comment:
The exist argument is needed regardless, because of the timeout - we need to know if we're waiting for a schema to exist, or to stop existing. Because of that, I think it makes more sense for it to throw on failure. Otherwise, we get double negatives like `assertSchema("subject", 10, false) => false` meaning that the schema exists but `assertSchema("subject", 10, true) => false` meaning that the schema doesn't exist.

Reply:
Ok, ok, I'm getting on that wave length.  Do we have any precedent here?

I don't see any `CompletableFuture<Boolean>` returns in `Client.java`, and the other methods which return `CompletableFuture<Void>` seem to be async requests.

Reply:
Hmm, yeah, in general I would expect CF<Void> to represent an async side effect. But I think this gives us the semantics we want, because as an assertion we don't actually care about the return type and we want failures to flow to the right/exceptional side the async chain.

```java
// make sure we have a foo topic and corresponding schemas
client
  .assertTopic("foo", true)
  .thenCompose((_) -> assertSchema("foo-key", true))
  .thenCompose((_) -> assertSchema("foo-value", true))
  .thenAccept((_) -> /* assertions passed; do other stuff */)
  .exceptionally((e) -> /* assertions failed; don't do other stuff and print reason */);
```

Reply:
Ok, I'm sold.  Approving the PR!  :shipit: